test/integration.js

```diff
@@ -0,0 +1,454 @@
+// Native
+const path = require('path')
+const { homedir } = require('os')
+const { URL } = require('url')
+
+// Packages
+const test = require('ava')
+const semVer = require('semver')
+const fkill = require('fkill')
+const { remove, pathExists, readJSON, writeJSON, readFile } = require('fs-extra')
+const execa = require('execa')
+const fetch = require('node-fetch')
+
+// Utilities
+const logo = require('../src/util/output/logo')
+const pkg = require('../package')
+const parseList = require('./helpers/parse-list')
+const removeDeployment = require('./helpers/remove')
+const prepareFixtures = require('./helpers/prepare')
+
+const binary = {
+  darwin: 'now-macos',
+  linux: 'now-linux',
+  win32: 'now-win.exe'
+}[process.platform]
+
+const binaryPath = path.resolve(__dirname, '../packed/' + binary)
+const fixture = name => path.join(__dirname, 'fixtures', 'integration', name)
+const deployHelpMessage = `${logo} now [options] <command | path>`
+const session = Math.random().toString(36).split('.')[1]
+
+// AVA's `t.context` can only be set before the tests,
+// but we want to set it within as well
+const context = {}
+
+const configDir = path.resolve(homedir(), '.now')
+
+const configFiles = {
+  auth: path.resolve(configDir, 'auth.json'),
+  config: path.resolve(configDir, 'config.json')
+}
+
+test.before(async () => {
+  let configContent
+
+  // Close the existing app
+  if (!process.env.CI) {
+    try {
+      await fkill('Now')
+    } catch (err) {}
```

Comment:
Why is it ok to swallow this error?

Reply:
Because its purpose is to close Now Desktop, so that the tests can easily replace the config and bring it back after the tests without Now Desktop noticing it.

Reply:
Why does `now-cli` know about Now Desktop?

Reply:
Now CLI doesn't know about Now Desktop and never will. However, since both share the same configuration file (which Now Desktop is watching and logging you out once it's gone), we need to close Now Desktop in the Now CLI >tests<, not from Now CLI's code itself.

Reply:
I don't think this is a good idea. Tests or not, the coupling should not exist

Reply:
@rauchg I'm open for different suggestions, but as far as i can see, it needs to. We cannot expect developers to close Now Desktop before they run the tests and we also cannot tell Now Desktop that we're running tests.

Reply:
Removed the local app closing and config restoring, as it's very dangerous.